### ISSUE TITLE
Ancestry table not expanded

### ISSUE THREAD

**Issue by Inbefortus** [openzim/mwoffliner#1491]
Kiwix Version: 3.4.4
ZIM: 2021-03 English Wikipedia

The table carrying ancestry is not unfolded.

Kiwix:

![Screenshot_20210705-092217_Kiwix](https://user-images.githubusercontent.com/71934042/124435889-bc009900-dd75-11eb-895d-a4a5d448ec40.jpg)

![Screenshot_20210705-092901_Kiwix](https://user-images.githubusercontent.com/71934042/124435893-bd31c600-dd75-11eb-924f-9622a8c92812.jpg)

Original English Wikipedia:

![Screenshot_20210705-092250_Samsung Internet](https://user-images.githubusercontent.com/71934042/124435944-cc187880-dd75-11eb-9002-97600efc3d3d.jpg)

![Screenshot_20210705-093032_Samsung Internet](https://user-images.githubusercontent.com/71934042/124435947-cd49a580-dd75-11eb-8e82-4df75530eee2.jpg)

**Comment by kelson42** [comment]
Which article is that?

**Comment by Inbefortus** [comment]
> Which article is that?

https://en.m.wikipedia.org/wiki/Louis_Antoine,_Duke_of_Enghien
and
https://en.m.wikipedia.org/w/index.php?title=Henry_III_of_England&oldid=1011571233&mobileaction=toggle_view_mobile

**Comment by Inbefortus** [comment]
@kelson42 I think this should rather be transferred to Kiwix Android as the ancestor table is observable in Kiwix JS PWA and not in  Android and mwoffliner did everything just right.

Kiwix JS PWA:

![Screenshot_20210727-092747_Samsung Internet](https://user-images.githubusercontent.com/71934042/127126379-b7d0e4ec-6c19-43b4-94d2-e51f3afe01d6.jpg)
![Screenshot_20210727-092847_Samsung Internet](https://user-images.githubusercontent.com/71934042/127126385-f76ad95d-d9f8-4245-b21f-8d3364626583.jpg)

**Comment by kelson42** [comment]
@Inbefortus Thx for this new info. But I will run the investigation first here and then triage the ticket accordingly. In general, the reader don't and shouldn't manipulate in any manner the content... and the way how things are displayed. Therefore, for the moment, I believe the ticket is at the right place.

**Comment by kelson42** [comment]
I guess it's fixed now. I will check.

**Comment by kelson42** [comment]
The problem seems to be still there for « Louis Antoine…. ».

**Comment by VadimKovalenkoSNF** [comment]
mwoffliner doesn't scrape article with the title `Louis_Antoine,_Duke_of_Enghien`. When I try to set this title to `--articleList`, it tries to scrape two different articles: `Louis_Antoine` and `Duke_of_Enghien`. Tried to encode title, but that didn't help.

**Comment by kelson42** [comment]
```
  --articleList               List of articles to include. Can be a comma sepera
                              ted list of titles or a local path or http(s) URL
                              to a file with one title (in UTF8) per line
```

**Comment by benoit74** [comment]
The problem is that this table has the "collapsible collapsed" CSS attribute. 

Something (JS) is responsible to handle this by adding the "show" button, but we miss it in the ZIM for some reason.

![Image](https://github.com/user-attachments/assets/d385d7ba-09e0-4cff-a4d6-dd20f47e514d)

If we remove the "collapsed" attribute (like JS would do) then everything is fine.

![Image](https://github.com/user-attachments/assets/05f129b7-a09e-4bf7-bb1e-7fff5e273fea)

We need to find which JS we are missing, are implemented our own version (not that hard anyway)

**Comment by benoit74** [comment]
See https://github.com/openzim/mwoffliner/issues/1911#issue-1903221546 for another case where the table is now uncollapsed by default (but we still want to add the show/hide button so that users can hide the table if they want to).

**Comment by Markus-Rost** [comment]
The JS module responsible for collapsible content included in MW itself is `jquery.makeCollapsible` controlling the `mw-collapsible` class (see [Manual:Collapsible elements](https://www.mediawiki.org/wiki/Manual:Collapsible_elements)). We would need to solve #2310 before we can consider loading this module, as it has multiple dependencies and we need a working `startup` script.
- `jquery.makeCollapsible`
  - `jquery.makeCollapsible.styles` (CSS module)
  - `mediawiki.util`
    - `jquery.client`
    - `web2017-polyfills`

Sadly even loading that module won't solve this specific issue, as Wikipedia (like many wikis) uses their own custom script for collapsible content (note `collapsible` class instead of `mw-collapsible`). The code might be in the general `site` module or in a gadget, so we wouldn't know which module we need to prioritize loading. Additionally any such module might make additional external requests (which we can't cache), besides just loading modules through the resource loader.

**Comment by benoit74** [comment]
I don't think we should be too afraid of having a module which attempts to load additional external requests. In all readers this is currently properly blocked. The only concern is hence whether the module works properly (or sufficiently properly) without these external resources or not. Of course it will cause some errors in console or somewhere else, but users don't care. If added value is here, I believe nobody should care much about few messages in a place mostly no-ones reads. 

Not even speaking about the fact that we know we can intercept these external resources loading. Either with the whole wombat.JS "stack" or with a dumbified version if we just want to return "nothing" on these external resources fetching.

**Comment by benoit74** [comment]
This is going to be solved by #2483

**Comment by benoit74** [comment]
I confirm this is now solved scraper-side. 

Reconfiguring recipe is tracked in https://github.com/openzim/zim-requests/issues/1687

With `--addModules site` CLI argument and current `main` branch:

```
docker run --rm --name mwoffliner_test -v $PWD/output:/output ghcr.io/openzim/mwoffliner:dev  mwoffliner --adminEmail=contact@kiwix.org --articleList=/output/list.tsv --customZimDescription="Test" --customZimTitle=Test --filenamePrefix=tests_en_mwoffliner --format=maxi --mwUrl=https://en.wikipedia.org --outputDirectory=/output --publisher=openZIM --web --addModules site
```

https://github.com/user-attachments/assets/0a6bbf53-4791-4b6f-aace-3c406742e741
